Add LFM elo computation

iFuSiiOnzZ opened the pull request:
- Get the entry list based on the player split
- Compute "magic" number used to compute the ELO win/loss threshold
- Get player car number
- Get ELO win/loss threshold
- Get current ELO win/loss based on position

NOTE: Needs to be tested
Based on: https://discord.com/channels/710171166601379881/1243706178589036605

iFuSiiOnzZ committed:
Add LFM elo computation

BradburyNO commented:
It isn't explained in the Discord post, and in case it isn't self-explanatory, I'll just add that when the calculation is multiplied with 0.8, this is elo multiplier (also called K-factor). This factor is found in the same place as the entrylist.

iFuSiiOnzZ committed:
Use race multiplayer from the LFM API

iFuSiiOnzZ commented:
Changed the multiplayer to read it from the API. All the other magic number as "divide by 100", "divide by 200", "1600", "LN(2)", etc. are always the same?

PS: Made a 20 min daily race and the computed ELO was correct.

BradburyNO commented:
Yes, those other numbers are constant.

RiddleTime commented on Race_Element.HUD.ACC/Overlays/Pitwall/OverlayLowFuelMotorsport/LowFuelMotorsportElo.cs:
```diff
@@ -0,0 +1,63 @@
+
+public class LowFuelMotorsportElo
+{
+    static readonly float LN2 = 0.69314718056f;
```
consider making these 2 floats a `const`, `readonly` is usually only used when the value gets initialized in the constructor.

RiddleTime commented on Race_Element.HUD.ACC/Overlays/Pitwall/OverlayLowFuelMotorsport/API/ApiObjects.cs:
```diff
@@ -70,3 +70,32 @@ internal readonly record struct User
+/// information used for compute the ELO and the position
+/// threshold where the player wins/losses ELO.
+/// </summary>
+public struct SplitEntry
```
Is the data immutable? if so `public readonly struct SplitEntry`

RiddleTime committed:
lfm hud

SplitEntry struct is now a readonly record struct with a primary constructor

iFuSiiOnzZ committed:
Use const instead of static readonly

iFuSiiOnzZ committed:
Use read only record

RiddleTime committed:
autoformat and refactor

RiddleTime committed:
Update LowFuelMotorsportOverlay.cs

add sealed keywords

RiddleTime committed:
Update LowFuelMotorsportSpeechSynthesizer.cs

refactor (sealed and primary constructor)

RiddleTime committed:
Update LowFuelMotorsportOverlay.cs

convert the list of speech jobs into a concurrent bag, please test this

RiddleTime committed:
Update LowFuelMotorsportSpeechSynthesizer.cs

- use pattern matching instead of the if else
- use TimeSpan.FromMinutes as it's easier to read than the constructor

iFuSiiOnzZ committed:
Merge branch 'RiddleTime:dev' into dev
